selfdrive/car/ecu_addrs.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+import capnp
+import time
+import traceback
+from typing import Dict, Optional, Set
+
+import cereal.messaging as messaging
+from panda.python.uds import SERVICE_TYPE
+from selfdrive.car import make_can_msg
+from selfdrive.boardd.boardd import can_list_to_can_capnp
+from selfdrive.swaglog import cloudlog
+
+
+def is_tester_present_response(msg: capnp.lib.capnp._DynamicStructReader) -> bool:
+  # ISO-TP messages are always padded to 8 bytes
+  # tester present response is always a single frame
+  if len(msg.dat) == 8 and 1 <= msg.dat[0] <= 7:
+    # success response
+    if msg.dat[1] == (SERVICE_TYPE.TESTER_PRESENT + 0x40):
+      return True
+    # error response
+    if msg.dat[1] == 0x7F and msg.dat[2] == SERVICE_TYPE.TESTER_PRESENT:
+      return True
+  return False
+
+
+def get_all_ecu_addrs(logcan: messaging.SubSocket, sendcan: messaging.PubSocket, bus: int, timeout: float = 1, debug: bool = True) -> Set[int]:
+  addr_list = [0x700 + i for i in range(256)] + [0x18da00f1 + (i << 8) for i in range(256)]
```

Comment:
Should we add an offset here since addresses at the top of the range won't be in the list?